internal/test/e2e/tinkerbellHardwareQueue.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+// Default timeout for Tink tests to poll for hardware.
+const hwPollingTimeout = 120 * time.Minute
+
+type hardwareQueue struct {
```

Comment:
I'm not sure a queue is the right way to frame this; it feels like it could be a semaphore. The order in which hardware is returned seems irrelevant. What inspired modeling this as a queue?

Reply:
We needed a in-memory data structure that is simple to implement and could provide a mechanism for hardware reservation. Although order doesn't really matter it could provide us extensibility if in future we need to have some control over the order in which these tests reserve the order! Also, there isn't necessarily any downside to keeping it as a queue for the current use case .?